Modules/_io/textio.c

```diff
@@ -1642,7 +1644,10 @@ _io_TextIOWrapper_write_impl(textio *self, PyObject *text)
 
     /* XXX What if we were just reading? */
     if (self->encodefunc != NULL) {
-        if (PyUnicode_IS_ASCII(text) && is_asciicompat_encoding(self->encodefunc)) {
+        if (PyUnicode_IS_ASCII(text) &&
+                // See bpo-43260
+                PyUnicode_GET_LENGTH(text) < self->chunk_size &&
+                is_asciicompat_encoding(self->encodefunc)) {
```

Comment:
I think it should be `<= self->chunk_size`, which ensures that `_textiowrapper_writeflush()` never has to allocate more than `chunk_size`. It doesn't need to be restricted to `chunk_size - 1`.